CI is broken due to nltk: Resource punkt_tab not found

CI is broken: Resource punkt_tab not found

https://github.com/huggingface/evaluate/actions/runs/10806684083/job/29975990875?pr=621
```
FAILED tests/test_metric_common.py::LocalModuleTest::test_load_metric_meteor - LookupError: 
**********************************************************************
  Resource punkt_tab not found.
  Please use the NLTK Downloader to obtain the resource:

  >>> import nltk
  >>> nltk.download('punkt_tab')
  
  For more information see: https://www.nltk.org/data.html

  Attempted to load tokenizers/punkt_tab/english/

  Searched in:
    - '/home/runner/nltk_data'
    - '/opt/hostedtoolcache/Python/3.8.18/x64/nltk_data'
    - '/opt/hostedtoolcache/Python/3.8.18/x64/share/nltk_data'
    - '/opt/hostedtoolcache/Python/3.8.18/x64/lib/nltk_data'
    - '/usr/share/nltk_data'
    - '/usr/local/share/nltk_data'
    - '/usr/lib/nltk_data'
    - '/usr/local/lib/nltk_data'
**********************************************************************
=========== 3 failed, 152 passed, 84 skipped, 39 warnings in 35.43s ============
```

--- a/setup.py
+++ b/setup.py
@@ -74,7 +74,7 @@ TESTS_REQUIRE = [
     "absl-py",
     "charcut>=1.1.1",  # for charcut_mt
     "cer>=1.2.0",  # for characTER
-    "nltk",  # for NIST and probably others
+    "nltk<3.9",  # for NIST and probably others; temporarily pin < 3.9 to avoid "Resource punkt_tab not found" (GH-622)
     "pytest",
     "pytest-datadir",
     "pytest-xdist",

